src/main/java/de/craftlancer/clcapture/CapturePoint.java

```diff
@@ -222,22 +188,83 @@ private void runInactive() {
     }
     
     private void runActive() {
+        boolean areMultipleClans = false;
+        int scoreMultiplier = 1;
+        List<UUID> inRegionList = new ArrayList<>();
         lastTime = LocalTime.now().toSecondOfDay();
         
-        timeMap.replaceAll((a, b) -> a.equals(currentOwner) ? b + 1 : Math.max(b - 1, 0));
+        //If a player is within the capturepoint region, add them to the list.
+        Bukkit.getOnlinePlayers().forEach(a -> {
+            if (isInRegion(a))
+                inRegionList.add(convertToOwner(a));
+        });
         
-        bar.setProgress(timeMap.getOrDefault(currentOwner, 0) / (double) type.getCaptureTime());
-        bar.setTitle(name + " - " + getOwnerName());
+        //See if there are multiple clans by comparing clans
+        areMultipleClans = checkAreMultipleClans(inRegionList, areMultipleClans);

```

Comment:
Change inRegionList to a Set, it doesn't allow duplicate entries. It would make checkAreMultipleClans obsolete since you can just check if inRegionList has a size >= 2 to detect multiple people.

Reply:
On further inspection, you care for the number of clan members so change it to a Map with key being the UUID and the value being the count of people of that clan.

`inRegionList.compute(convertToOwner(a), (b, c) -> b == null ? 1 : c + 1)`